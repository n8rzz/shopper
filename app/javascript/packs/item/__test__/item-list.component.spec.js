import React from 'react';
import {shallow} from 'enzyme';

import {ItemList} from '../item-list.component';
import {
    departmentMapMock,
    itemListMock
} from './__mock__/item.mocks';

describe('ItemList', () => {
    test('renders correctly with valid props', () => {
        const component = shallow(<ItemList
            csrf={'3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=='}
            departmentMap={departmentMapMock}
            itemList={itemListMock}
        />);

        expect(component).toMatchSnapshot();
    });

    describe('when passed no items', () => {
        test('returns `null`', () => {
            const component = shallow(<ItemList
                csrf={'3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=='}
                departmentMap={departmentMapMock}
                itemList={[]}
            />);

            expect(component).not.toBeUndefined();
        });
    })
});
