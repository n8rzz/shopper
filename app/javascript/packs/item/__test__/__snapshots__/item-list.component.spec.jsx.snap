// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList renders correctly with valid props 1`] = `
<ul
  className="vlist"
>
  <Item
    csrf="3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=="
    departmentId={5}
    departmentName="Meat"
    itemId={15}
    itemName="Agave Syrup"
    key="item-0"
  />
  <Item
    csrf="3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=="
    departmentId={3}
    departmentName="Bakery"
    itemId={13}
    itemName="Agave Syrup"
    key="item-1"
  />
</ul>
`;
